Allow upserting data via bulk transaction requests

edkeeble opened the pull request:
**Description:**

[Along with this PR on stac-fastapi](https://github.com/stac-utils/stac-fastapi/pull/614), this change allows a client to make use of the`upsert_items` function by way of the Bulk Transactions extension.

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

gadomski reviewed (changes requested):
Actually, can you add a test case demonstrating the upsertion? Thanks! 🙇🏼

edkeeble commented:
> Actually, can you add a test case demonstrating the upsertion? Thanks! 🙇🏼

@gadomski Yep, added a couple of tests to confirm the expected behavior. I also modified the response message to reflect the provided insert method (while retaining "added" for the insert method, for the purposes of backward compatibility). Not sure how you feel about that, but happy to revert it to always return "added".

edkeeble commented:
@gadomski Yes, please. I'll put a separate PR up for the release.

edkeeble commented:
@gadomski https://github.com/stac-utils/stac-fastapi-pgstac/pull/68
